### PR TITLE
Added google analytics tracking with yarn ga.

### PR DESCRIPTION
**Pull request by abhikhatri67**
# Description

Google analytics support in react play.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by abhikhatri67** [comment]
@atapas if this looks fine to you. You can send the tracking id, or do you want to enter it manually so that you can test it?

**Comment by atapas** [comment]
@abhikhatri67 Kindly resolve the conflicts.

**Review by atapas — CHANGES_REQUESTED** [review]

**Comment by abhikhatri67** [comment]
> > @atapas if this looks fine to you. You can send the tracking id, or do you want to enter it manually so that you can test it?
> 
> @abhikhatri67 Please share your email, I shall send it.
> 
> Also, is the code will take care of analytics for all the routes?

Not sure about this @atapas, have to test it.

File: src/index.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ import React, { useState } from "react";
 import { createRoot } from "react-dom/client";
 import reportWebVitals from "reportWebVitals";
 
+require('dotenv').config()
```
Comment (atapas) [inline src/index.js:8]: You don't need this. It will fail the build.

File: src/App.js [hunk 2]
```diff
@@ -1,7 +1,15 @@
-import "App.css";
 import { Outlet } from "react-router-dom";
+import ReactGA from 'react-ga';
+
+import "App.css";
+
+ReactGA.initialize(process.env.REACT_APP_GA_TRACKING_ID);
```
Comment (atapas) [inline src/App.js:6]: I have added the env variable with this key in vercel. So when the build passes, the analytics will work.

**Review by atapas — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (abhikhatri67) [reply src/index.js]: Done.

**Comment by atapas** [comment]
The build fails:

image

**Comment by abhikhatri67** [comment]
> > > > @atapas if this looks fine to you. You can send the tracking id, or do you want to enter it manually so that you can test it?
> > > 
> > > 
> > > @abhikhatri67 Please share your email, I shall send it.
> > > Also, is the code will take care of analytics for all the routes?
> > 
> > 
> > Not sure about this @atapas, have to test it.
> 
> @abhikhatri67 If you could make the build pass will be great. We need to test it out. Please gimme your email id so that I can add you as an admin to the GA board to test.

@atapas sure will do that, and here is my email id abhikhatri67@gmail.com

**Comment by abhikhatri67** [comment]
Strange my build is failing due to the puppeteer I think.

Screenshot 2022-05-03 at 2 52 11 PM

**Comment by atapas** [comment]
> Strange my build is failing due to the puppeteer I think.
> 
> Screenshot 2022-05-03 at 2 52 11 PM

That's error is intermittent, we will fix.

For the time being, rin the yarn build again when it fails. It will go thru in the next attempt.

**Comment by abhikhatri67** [comment]
> 

Hey, @atapas build is running fine for me, and it's not throwing the error. 

Can you check it again?

Screenshot 2022-05-07 at 8 27 36 PM

**Comment by atapas** [comment]
> > > > > @atapas if this looks fine to you. You can send the tracking id, or do you want to enter it manually so that you can test it?
> > > > 
> > > > 
> > > > @abhikhatri67 Please share your email, I shall send it.
> > > > Also, is the code will take care of analytics for all the routes?
> > > 
> > > 
> > > Not sure about this @atapas, have to test it.
> > 
> > 
> > @abhikhatri67 If you could make the build pass will be great. We need to test it out. Please gimme your email id so that I can add you as an admin to the GA board to test.
> 
> @atapas sure will do that, and here is my email id [abhikhatri67@gmail.com](mailto:abhikhatri67@gmail.com)

I will add you tonight in google analytics dashboard as admin. You can test it from the preview url after that.

**Comment by atapas** [comment]
> Hey, @atapas build is running fine for me, and it's not throwing the error.
> 
> Can you check it again?
> 
> Screenshot 2022-05-07 at 8 27 36 PM

Build is passing on vercel too.

**Comment by atapas** [comment]
@abhikhatri67 You are added and got all the power to set things up

image

Please let me know if you need any help from me.

**Comment by atapas** [comment]
@abhikhatri67 Could you test this one?

**Comment by abhikhatri67** [comment]
> @abhikhatri67 Could you test this one?

It's not working @atapas, I need your help in this one.

**Comment by atapas** [comment]
> > @abhikhatri67 Could you test this one?
> 
> It's not working @atapas, I need your help in this one.

No worries, will do. Now at medical leave, will do sooner I’m by.

**Comment by atapas** [comment]
Hello @abhikhatri67 

We will be moving the react-play project from the `atapas` account to an organization to help manage the project in a better way. The new organization URL is: https://github.com/reactplay.

> **The movement will take place between 12:30 - 14:00 hrs IST(7:00 am - 8:30 am GMT), 30th May 2022. Please refrain from pushing any changes between this time.**

We will inform you when the movement is complete. You need to make a few minimal configuration changes after that. We will let you know.

**Comment by atapas** [comment]
> Hello @abhikhatri67
> 
> We will be moving the react-play project from the `atapas` account to an organization to help manage the project in a better way. The new organization URL is: https://github.com/reactplay.
> 
> > **The movement will take place between 12:30 - 14:00 hrs IST(7:00 am - 8:30 am GMT), 30th May 2022. Please refrain from pushing any changes between this time.**
> 
> We will inform you when the movement is complete. You need to make a few minimal configuration changes after that. We will let you know.

The migration is complete. You can find the project here: https://github.com/reactplay

You no need to reclone the project. You just need to set the correct upstream. Here is the link: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-for-a-fork

**Comment by atapas** [comment]
This PR is no more valid as we have integrated with the Umami Analytics.

**Pull request closed without merging**